tink-envelope-encryption-sample/3-reencrypt-symetric-to-envelope/main.tf

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "null_resource" "decrypt_current_file" {
+
+  triggers = {
```

Comment:
How do triggers work here? Genuine question, still learning terraform :)

Reply:
No problem, it is a good question.
Basically, `triggers` are used in `null_resources` in order to let terraform know when it is supposed to replace the current resource, or in this case, to run the command again. In other words, the `gcloud kms decrypt` command should be triggered (when a`terraform apply` is run, and) if detects a change in `var.current_encrypted_file_path` or `var.rotate_encrypted_file_path` after the first apply.

In summary:
- The first tf apply will always run the command;
- If triggers doesn't exist, terraform will never run the command again after the first tf apply;
- If triggers exists, terraform will run the command again only if a trigger variable were changed.

[From docs](https://registry.terraform.io/providers/hashicorp/null/latest/docs/resources/resource):
"The triggers argument allows specifying an arbitrary set of values that, when changed, will cause the resource to be replaced."